### PR TITLE
Some 💜 to struct code generation

### PR DESCRIPTION
**Pull request by hugopl**
## The problem
Struct code generation in gi-crystal have a lot of problems

- Generate heap allocation when not needed.
- Isn't flexible enough for the different ways different structs are meant to be used out there.

Trying to use [Pango](https://docs.gtk.org/Pango/index.html) and [Graphene](http://ebassi.github.io/graphene/docs/ch01.html) bindings was frustrating

- [PangoAttribute](https://docs.gtk.org/Pango/struct.Attribute.html) has a field named `klass` that must not be present in the bindings.
- PangoAttribute crash, since must be freed with `g_free`.
- PangoFontMetrics crash, probably same issue.
- PangoFontMetrics must be immutable like it is in other bindings.
- Basically all [Graphene](http://ebassi.github.io/graphene/docs/ch01.html) types must be binded as Crystal structs to avoid the heap allocation overhead.

## Progress
- [x] Refactor binding.yml format so it can give more information about the types, then we can add more options to tell how the struct bindings should be generated.
- [x] ~Allow immutable struct bindings.~
- [x] Bind POD structs as Crystal structs.
- [x] Allow removal of struct fields in bindings.
- [x] Allow different strategies for struct binding generation.

## Definition of done

Make Pango and Graphene work as expected.

Fix https://github.com/hugopl/gi-crystal/issues/51

### CODE REVIEW

**Comment by hugopl** [comment]
I added code to make structs that have no pointers be binded as Crystal structs, but one test case still failing.

What's should the binding do for `Struct#point_ptr=` in the use case bellow?
```Crystal
lib LibTest
  struct Point
    x : Int32
    y : Int32
  end

  struct Struct
    point_ptr : Pointer(LibTest::Point)
  end
end

s = Test::Struct.new
p = Test::Point.new
# How this should be exposed to Crystal!?
s.point_ptr = p
```

Possible "solutions":
- Print a warning and let the binding author find a way to solve this.
- Create a `PointRef` heap allocated object, use its pointer and let the GC take care of it.

I didn't see this use case in the wild yet, so I tend to choose the first "solution" then do the second or other better one when someone deal with this situation with a real binding.
